tests/pr_test.go

```diff
@@ -46,14 +47,17 @@ func setupOptionsBasicExample(t *testing.T, prefix string) *testhelper.TestOptio
 		ResourceGroup:    resourceGroup,
 		CloudInfoService: sharedInfoSvc, // use pointer to shared info svc to keep track of region selections
 		DefaultRegion:    "us-south",
-		TerraformVars: map[string]interface{}{
-			"transit_gateway_name":      fmt.Sprintf("%s-%s", prefix, "tg"),
-			"vpc_connections":           vpcConnections,
-			"classic_connections_count": 0,
-		},
-		TerraformDir: basicExampleTerraformDir,
+		TerraformDir:     basicExampleTerraformDir,
 	})
 
+	terraformVars := map[string]interface{}{
+		"transit_gateway_name":      fmt.Sprintf("%s-%s", options.Prefix, "tg"),
+		"vpc_connections":           vpcConnections,
+		"classic_connections_count": 0,
+	}
+
+	maps.Copy(options.TerraformVars, terraformVars)
```

Comment:
I think you can use `options.TerraformVars = map[string]interface{}{` here instead of using the `maps.Copy()` function

Reply:
Is this what you meant?  ```options.TerraformVars = map[string]interface{}{options.TerraformVars, terraformVars}```

Reply:
No, `options` is being set on line 44, so now you append `TerraformVars` into `options` using:
```
options.TerraformVars = map[string]interface{}{
                "transit_gateway_name":      fmt.Sprintf("%s-%s", options.Prefix, "tg"),
		"vpc_connections":           vpcConnections,
		"classic_connections_count": 0,
	}
```

Reply:
Ah okay, I tried that and it overwrites everything that was previously in options.TerraformVars.

Reply:
TestOptionsDefaultWithVars supports merging default TerraformVars with extra TerraformVars received in input
For example have a look here for this approach
https://github.ibm.com/GoldenEye/ocp-service-mesh-module/blob/36be3e3079607aad0814326990a02f70a4c24130/tests/pr_test.go#L103
https://github.ibm.com/GoldenEye/ocp-service-mesh-module/blob/36be3e3079607aad0814326990a02f70a4c24130/tests/pr_test.go#L48

Reply:
The issue with that is that the options.Prefix hasn't been set yet, and we want the Prefix with the random bit that testhelper.TestOptionsDefaultWithVars sets up. It has to be merged after the function has been run.